Using kubectl to obtain public IP for guestbook service

satnam6502 opened the pull request:
Related to #4577 this changes the Guestbook example documentation to use `kubectl` to obtain the public IP of the guestbook service rather than the `gcloud` command (which after #4577 will result in a forwarding rule which is no longer exactly the name of the service).

satnam6502 committed:
Using kubectl to obtain public IP for guestbook service

satnam6502 commented:
After months of happily making commits why do I now get these CLA messages?
I filled in form a few days ago. I guess that was not enough.
As a Google employee, what form should I fill in?

yujuhong commented:
From @lavalamp: It's important to use your @google.com email address in your commits (edit .git/config) to make the cla bot happy.

The git config instructions: https://help.github.com/articles/setting-your-email-in-git

satnam6502 commented:
I've changed my email to `satnam@google.com` for commits from my kubernetes repo.

satnam6502 commented:
Do I now need to fill out https://cla.developers.google.com/clas/new?kind=KIND_CORPORATE&domain=DOMAIN_GOOGLE ? If yes, then what the value values for contributor group and primary point of contact and authorized signer? Thanks.

lavalamp commented:
I don't think googlers need to fill that out.

satnam6502 commented:
So what else do I need to do to stop getting these googlebot annoying messages?

lavalamp commented:
We won't know if your change to your @google.com email worked until you make another PR-- the email is embedded in the commit, which you made previously to changing the email-- right?

yujuhong commented:
Also, @satnam6502, I still can't find your name in the Google organization member list. Maybe it takes time to propagate/approve? Do you see a Google organization logo on your github page?

satnam6502 commented:
@yujuhong : I think I am sorted now.

yujuhong commented:
@satnam6502, glad to hear that!

I saw the member list from https://github.com/orgs/google/people

satnam6502 commented:
Oh, I am not on that member list. Neither are you :-) And I can't spot any other Googlers on that list. Should I be on it?

satnam6502 commented:
Ping?

jlowdermilk commented on examples/guestbook-go/README.md:
````diff
@@ -159,20 +159,34 @@ redis-master            <none>                                    name=redis,rol
+To play with the service itself, find the external IP of the load balancer as shown below by extracting a value for `publicIPs` (in this example `107.178.218.166`) and then visit `http://<ip>:3000`.
+
+```shell
+$ cluster/kubectl.sh get services guestbook -o json
````
Optional: if you use a template as the output, you can extract just the publicIPs. Try
```
kubectl get services guestbook -o template --template='{{index . "publicIPs"}}'
```

satnam6502 replied:
Excellent, I was wondering how to do that. I'll update this PR.

satnam6502 commented:
Updated to use @jlowdermilk template that prints out just the public IPs and I also added a section suggesting the explicit deletion of the service which will clean up any project level resources like forwarding rules and target pools. PTAL, thank you.

jlowdermilk commented on examples/guestbook-go/README.md:
````diff
@@ -189,6 +184,13 @@ For details about limiting traffic to specific sources, see the [GCE firewall do
 
+You should delete the service which will remove any associated resources that were created e.g. load balancers, forwarding rules and target pools.
+```shell
+$ cluster/kubectl.sh delete service guestbook
````
Nit: I think you can do a delete -f on the guestbook directory to cleanup all the resources, including the service.

jlowdermilk commented:
LGTM for the readme change, but you appear to have outdated gendocs output in kubectl.md. Try `make clean` followed by `hack/run-gendocs.sh`

satnam6502 commented:
@jlowdermilk : I've fixed the docs issues, thanks, and I've changed the cleanup command to do a bulk delete as you suggest. I think it's good to apply the LGTM label (when you are ready) to make it easier for people to scan PRs that are LGTM-ed but not committed yet. Thanks.

jlowdermilk commented:
LGTM. shippable failure is flake. merging.
